src/components/search/ToggleThreeButton.tsx

```diff
@@ -0,0 +1,46 @@
+import { ToggleButton, ToggleButtonGroup, Typography } from '@mui/material'
+import { Fragment } from 'react'
+
+import ToggleThreeButtonType from '../../types/ToggleThreeButton'
+//TODO: update to the correct prop type
+const ToggleThreeButton = (prop: ToggleThreeButtonType) => {
```

Comment:
```suggestion
const ToggleThreeButton = (prop: ToggleThreeButtonType) => {
```
I suggest we make it more generic (not only for three buttons), we pass on a prop called "range" that we can iterate through to display ToggleButtons using the map function : 
```
[...Array(3).keys()];
 => [0, 1, 2]
```
That way you can do : 
```
[...Array(range).keys()].map((index) => {
    const value = index+1;
    return <ToggleButton
          value={value}
          aria-label="left aligned"
          sx={prop.ToggleButtonStyle}
          name={value}
        >
          <Typography id={value}>{value}</Typography>
        </ToggleButton>;
})
```

Reply:
Yes, great idea.